typing/cmt_format.ml

```diff
@@ -164,6 +164,12 @@ let record_value_dependency vd1 vd2 =
 let save_cmt filename modname binary_annots sourcefile initial_env sg =
   if !Clflags.binary_annotations && not !Clflags.print_types then begin
     let imports = Env.imports () in
+    let flags =
+      if !Clflags.recursive_types then [Cmi_format.Rectypes] else []
+    in
+    let flags =
+      if !Clflags.opaque then Cmi_format.Opaque :: flags else flags
+    in
```

Comment:
This code is duplicated twice, here and in Env. Would it make sense to factorize it into a `Cmi_format.global_pers_flags ()` function?

Also with my nitpicking hat on, I think you could factorize this as

```ocaml
  []
  @ (if !Clflags.recursive_types then [Cmi_format.Rectypes] else [])
  @ (if !Clflags.opaque then Cmi_format.Opaque :: flags else flags)
```